src/controllers/publish.ts

```diff
@@ -446,18 +446,23 @@ export const measurePreview = async (req: Request, res: Response, next: NextFunc
         }
 
         const dataPreview = await req.pubapi.getMeasurePreview(res.locals.dataset.id);
+        let langCol = -1;
+        if (dataPreview.headers.find((header) => header.name.toLowerCase().indexOf('lang') > -1)) {
+            langCol = dataPreview.headers.findIndex((header) => header.name.toLowerCase().indexOf('lang') > -1);
+        }
```

Comment:
You want the index of the header column that contains the string 'lang'?

Is this the same thing (findIndex() returns -1 if not found)?

```suggestion
        const langCol = dataPreview.headers.findIndex((header) => header.name.toLowerCase().includes('lang'))
```